client/src/main/kotlin/no/nav/su/se/bakover/client/ExpiringTokenResponse.kt

```diff
@@ -0,0 +1,28 @@
+package no.nav.su.se.bakover.client
+
+import org.json.JSONObject
+import java.time.LocalDateTime
+
+open class ExpiringTokenResponse(
+    private val json: JSONObject
+) {
+    val accessToken: AccessToken = AccessToken(json.getString("access_token"))
+    private val expiresIn: Int = json.getInt("expires_in")
+    private val expirationTime = LocalDateTime.now().plusSeconds(expiresIn - 20L)
+
+    companion object {
+        fun isValid(token: ExpiringTokenResponse?): Boolean {
+            return when (token) {
+                null -> false
+                else -> !token.isExpired()
+            }
+        }
```

Comment:
Flisespikkeri (og ikke et must), men synes det er nice å bruke optional chaining når vi har det

```suggestion
        fun isValid(token: ExpiringTokenResponse?): Boolean = !token?.isExpired() ?: false
```

Reply:
Denne må bli `fun isValid(token: ExpiringTokenResponse?): Boolean = token?.isExpired() == false`, men tenker den er mer leselig